pkg/reconciler/configuration/configuration.go

```diff
@@ -186,58 +186,55 @@ func (c *Reconciler) reconcile(ctx context.Context, config *v1alpha1.Configurati
 	default:
 		return fmt.Errorf("unrecognized condition status: %v on revision %q", rc.Status, revName)
 	}
-	if rc != nil && rc.Status == corev1.ConditionTrue {
-		old, new := config.Status.LatestReadyRevisionName, lcr.Name
-		config.Status.SetLatestReadyRevisionName(lcr.Name)
-		if old != new {
-			c.Recorder.Eventf(config, corev1.EventTypeNormal, "LatestReadyUpdate",
-				"LatestReadyRevisionName updated to %q", new)
-		}
-	} else {
-		if err = c.findAndSetLatestReadyRevision(config); err != nil {
-			return fmt.Errorf("failed to find and set latest ready revision: %w", err)
-		}
+
+	lcrReady := rc != nil && rc.Status == corev1.ConditionTrue
+	if err = c.findAndSetLatestReadyRevision(config, lcrReady); err != nil {
+		return fmt.Errorf("failed to find and set latest ready revision: %w", err)
 	}
 
 	return nil
 }
 
-// findAndSetLatestReadyRevision finds the last ready revision and sets LatestReadyRevisionName to it
-func (c *Reconciler) findAndSetLatestReadyRevision(config *v1alpha1.Configuration) error {
+// findAndSetLatestReadyRevision finds the last ready revision and sets LatestReadyRevisionName to it.
+// lcrReady indicates whether the latest created revision is ready or not.
+func (c *Reconciler) findAndSetLatestReadyRevision(config *v1alpha1.Configuration, lcrReady bool) error {
 	sortedRevisions, err := c.getSortedCreatedRevisions(config)
 	if err != nil {
 		return err
 	}
 	for _, rev := range sortedRevisions {
 		if rev.Status.IsReady() {
 			// No need to update latest ready revision in this case
-			if rev.Name == config.Status.LatestReadyRevisionName {
+			if rev.Name == config.Status.LatestReadyRevisionName && !lcrReady {
```

Comment:
This is for the [failed revision recovers use case](https://github.com/knative/serving/blob/master/pkg/reconciler/configuration/configuration_test.go#L344). We still need to call `SetLatestReadyRevisionName(...)` because it marks the configuration ready to true.

Reply:
I think threading through the added state probably makes things more confusing than its worth given that we save a tiny bit of redundancy in the SetLatestReadyRevision call when it's already true?